pkgs/development/python-modules/rchitect/default.nix

```diff
@@ -0,0 +1,52 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, cffi
+, six
+, pytest
+, pytest-mock
+, R
+, rPackages }:
+
+buildPythonPackage rec {
+  pname = "rchitect";
+  version = "0.3.40";
+
+  src = fetchFromGitHub {
+    owner = "randy3k";
+    repo = pname;
+    rev = "releases/tags/v${version}";
+    sha256 = "yJMiPmusZ62dd6+5VkA2uSjq57a0C3arG8CgiUUHKpk=";
+  };
+
+  postPatch = ''
+    substituteInPlace setup.py --replace '"pytest-runner"' ""
+  '';
+
+  propagatedBuildInputs = [
+    cffi
+    six
+  ] ++ (with rPackages; [
+    reticulate
+  ]);
+
+  nativeCheckInputs = [
+    pytest
+    pytest-mock
+    R
+  ];
+
+  checkPhase = ''
+    export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:${pkgs.R}/lib/R/lib
+    pytest
```

Comment:
```suggestion
  preCheck = ''
    export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:${pkgs.R}/lib/R/lib
```

Reply:
This won't work. `pytestCheckHook` does not pass `LD_LIBRARY_PATH` to `pytest`. `pytest` has to be called directly through `checkPhase` for this to work.

Reply:
> `pytestCheckHook` does not pass `LD_LIBRARY_PATH` to `pytest`.

Are you sure?
Anyway, we shouldn't have to set `LD_LIBRARY_PATH`. Maybe we should hardcode the path to those libraries using a patch.

Reply:
This is only required during testing. This library is not used when `radian` is used normally. I started doing this only because the author sets this for their automatic tests.

Reply:
Going to mark this as resolved as well, I don't think there's much to be gained by changing this.